src/include/partitioning/ep-dialogs.rb

```diff
@@ -445,7 +444,7 @@ def MiniWorkflowStepFormatMount(orig_data, creating: false)
       #not there in RAID/LVM/loop configuration (#483789)
       ChangeWidgetIfExists(:do_not_format_attachment, :Enabled, !do_format)
 
-      _EnableDisableFsOpts.call(used_fs) if do_format
```

Comment:
That function is a misnomer: It doesn't only enable or disable the "filesystem options" button, but also the "encrypt" check box. And this needs to be called in **every** case, not only if the user wants to format a partition. Otherwise that check box never gets enabled for partition type LVM (LVM physical volume) or RAID.

That enabling/disabling code in that dialog is a big mess anyway, but now it becomes a little bit more usable. There is no reason not to check enabled/disabled state after a user action, even more so since in that dialog everything depends on everything else.